source/utilities.tex

```diff
@@ -16908,6 +16939,27 @@
  Each type in the template parameter pack \tcode{T} shall be
  complete, \cv{}~\tcode{void}, or an array of unknown bound. \\ \rowsep
 
+\indexlibrary{\idxcode{basic_common_reference}}%
+\tcode{template <class, class,}
+ \hspace*{2ex}\tcode{template <class> class,}
+ \hspace*{2ex}\tcode{template <class> class>}
+ \tcode{struct}
+ \hspace*{2ex}\tcode{basic_common_reference;}
+ &
+ A program may specialize this trait if at least one
+ template parameter in the specialization depends on a user-defined
+ type. If there is a member \tcode{type}, it shall be a \grammarterm{typedef-name}.
```

Comment:
(not editorial) This seems an overspecification; certainly a user could define the member "type" as a nested class (which is not a typedef-name). The difference is, I believe, unobservable.

Reply:
Technically, it's observable in some cases with an _elaborated-type-specifier_.

Reply:
I'd be happy to change this to "If there is a member `\tcode{type}`, it shall denote a type." if we're willing to call that change editorial-ish.

Reply:
@timsong-cpp is right, such a change would be observable so can't be done editorially. @CaseyCarter, could you please check whether there is already an LWG issue for that, and file one if not?

Reply:
Yes I will. I have a set of notes on all the comments here so I can track which changes I've accomplished. There are six entries now for issues to file post-merge.